x-pack/plugins/case/server/connectors/case/index.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { curry } from 'lodash';
+
+import { KibanaRequest } from 'kibana/server';
+import { ActionTypeExecutorResult } from '../../../../actions/common';
+import { CasePatchRequest, CasePostRequest } from '../../../common/api';
+import { createCaseClient } from '../../client';
+import { CaseExecutorParamsSchema, CaseConfigurationSchema } from './schema';
+import {
+  CaseExecutorParams,
+  CaseExecutorResponse,
+  ExecutorSubActionCreateParams,
+  ExecutorSubActionUpdateParams,
+  ExecutorSubActionAddCommentParams,
+  CaseActionType,
+  CaseActionTypeExecutorOptions,
+} from './types';
+
+import { GetActionTypeParams } from '..';
+
+const supportedSubActions: string[] = ['create', 'update', 'addComment'];
+
+// action type definition
+export function getActionType({
+  logger,
+  caseService,
+  caseConfigureService,
+  userActionService,
+}: GetActionTypeParams): CaseActionType {
+  return {
+    id: '.case',
+    minimumLicenseRequired: 'gold',
+    name: 'Case',
+    validate: {
+      config: CaseConfigurationSchema,
+      params: CaseExecutorParamsSchema,
+    },
+    executor: curry(executor)({ logger, caseService, caseConfigureService, userActionService }),
+  };
+}
+
+// action executor
+
+async function executor(
+  { logger, caseService, caseConfigureService, userActionService }: GetActionTypeParams,
+  execOptions: CaseActionTypeExecutorOptions
+): Promise<ActionTypeExecutorResult<CaseExecutorResponse | {}>> {
+  const { actionId, params, services } = execOptions;
+  const { subAction, subActionParams } = params as CaseExecutorParams;
+  let data: CaseExecutorResponse | null = null;
+
+  const { savedObjectsClient } = services;
+  const caseClient = createCaseClient({
+    savedObjectsClient,
+    request: {} as KibanaRequest,
+    caseService,
+    caseConfigureService,
+    userActionService,
+  });
+
+  if (!supportedSubActions.includes(subAction)) {
+    const errorMessage = `[Action][Case] subAction ${subAction} not implemented.`;
+    logger.error(errorMessage);
+    throw new Error(errorMessage);
+  }
+
+  if (subAction === 'create') {
+    const createParams = subActionParams as ExecutorSubActionCreateParams;
+
+    data = await caseClient.create({ theCase: createParams as CasePostRequest });
```

Comment:
In short, `createParams = subActionParams as ExecutorSubActionCreateParams as CasePostRequest`... why cast to `ExecutorSubActionCreateParams`?

Reply:
I like the way you handled it for `addComment`
```
  if (subAction === 'create') {
    const createParams = subActionParams as ExecutorSubActionCreateParams;
    const { description, title, tags, connector } = createParams;

    data = await caseClient.create({ theCase: { description, title, tags, connector } });
  }
```

Reply:
You are right. There is no need to cast it `ExecutorSubActionCreateParams`.